src/cloudevents/core/bindings/http.py

```diff
@@ -0,0 +1,297 @@
+    and applies percent-encoding for non-ASCII and special characters per RFC 3986.
+
+    :param value: The attribute value to encode
+    :return: Percent-encoded string suitable for HTTP headers
+    """
+    if isinstance(value, datetime):
+        str_value = value.isoformat()
+        if str_value.endswith("+00:00"):
+            str_value = str_value[:-6] + "Z"
+        return quote(str_value, safe="")
+
+    return quote(str(value), safe="")
+
+
+def _decode_header_value(attr_name: str, value: str) -> Any:
+    """
+    Decode a CloudEvent attribute value from an HTTP header.
+
+    Applies percent-decoding and special parsing for the 'time' attribute
+    (converts to datetime object using RFC 3339 parsing).
+
+    :param attr_name: The name of the CloudEvent attribute
+    :param value: The percent-encoded header value
+    :return: Decoded value (datetime for 'time' attribute, string otherwise)
+    """
+    decoded = unquote(value)
+
+    if attr_name == "time":
+        return isoparse(decoded)
+
+    return decoded
+
+
+def to_binary(event: BaseCloudEvent, event_format: Format) -> HTTPMessage:
+    """
+    Convert a CloudEvent to HTTP binary content mode.
+
+    In binary mode, CloudEvent attributes are mapped to HTTP headers with the 'ce-' prefix,
+    except for 'datacontenttype' which maps to the 'Content-Type' header. The event data
+    is placed directly in the HTTP body.
+
+    Example:
+        >>> from cloudevents.core.v1.event import CloudEvent
+        >>> from cloudevents.core.formats.json import JSONFormat
+        >>>
+        >>> event = CloudEvent(
+        ...     attributes={"type": "com.example.test", "source": "/test"},
+        ...     data={"message": "Hello"}
+        ... )
+        >>> message = to_binary(event, JSONFormat())
+        >>> # message.headers = {"ce-type": "com.example.test", "ce-source": "/test", ...}
+        >>> # message.body = b'{"message": "Hello"}'
+
+    :param event: The CloudEvent to convert
+    :param event_format: Format implementation for data serialization
+    :return: HTTPMessage with ce-prefixed headers and event data as body
+    """
+    headers: Dict[str, str] = {}
+    attributes = event.get_attributes()
+
+    for attr_name, attr_value in attributes.items():
+        if attr_value is None:
+            continue
+
+        if attr_name == "datacontenttype":
+            headers[CONTENT_TYPE_HEADER] = str(attr_value)
+        else:
+            header_name = f"{CE_PREFIX}{attr_name}"
+            headers[header_name] = _encode_header_value(attr_value)
+
+    data = event.get_data()
+    datacontenttype = attributes.get("datacontenttype")
+    body = event_format.write_data(data, datacontenttype)
+
+    return HTTPMessage(headers=headers, body=body)
+
+
+def from_binary(
+    message: HTTPMessage,
+    event_format: Format,
+    event_factory: Callable[
+        [Dict[str, Any], Optional[Union[Dict[str, Any], str, bytes]]], BaseCloudEvent
+    ],
```

Comment:
I guess this should do it as well, right?
```suggestion
    event_factory: Callable[
        [dict[str, Any], dict[str, Any] | str | bytes | None], BaseCloudEvent
    ],
```